src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -1,182 +1,279 @@
-    const [myId, setMyId] = React.useState('')
-    // const [isFirstTimeOpen, setIsFirstTimeOpen] = React.useState(true)
-    const [discovery, discoveryDispatch] = React.useContext(Context).discovery
-    const [following, followingDispatch] = React.useContext(Context).following
-
-    const isReady = useIsReady()
-
-    // const { domains } = following
-    const domains = discovery.initialDomains
-
-    React.useEffect(() => {
-        const parseToken = async () => {
-            const id = await getUserId();
-            if (id) {
-                setMyId(id);
-            }
-        };
-        parseToken();
-    }, []);
-
-    // React.useEffect(() => {
-    //     if(followedDomains.length > 0 || unfollowedDomains.length > 0) setIsFirstTimeOpen(false)
-    // },[ followedDomains, unfollowedDomains ])
-
-    const __handleOnPressDomain = (item) => {
-        const navigationParam = {
-            item: {
-                content: {
-                    domain_page_id: item.domain_page_id
-                },
-                domain: {
-                    image: item.logo,
-                    credderScore: item?.credder_score,
-                },
-                og: {
-                    domain: item.domain_name,
-                    domainImage: item.logo,
-                }
-            }
-        }
+const DomainFragment = ({
+  isLoadingDiscoveryDomain,
+  isFirstTimeOpen,
+  followedDomains,
+  unfollowedDomains,
+  setFollowedDomains,
+  setUnfollowedDomains,
+  setSearchText,
+  setIsFirstTimeOpen,
+  fetchData = () => {},
+  searchText,
+  withoutRecent = false
+}) => {
+  const navigation = useNavigation();
+  const [myId, setMyId] = React.useState('');
+  const [discovery, discoveryDispatch] = React.useContext(Context).discovery;
+  const [, followingDispatch] = React.useContext(Context).following;
 
-        navigation.push('DomainScreen', navigationParam)
-    }
+  const isReady = useIsReady();
 
-    const __handleFollow = async (from, willFollow, item, index) => {
-        if (from === FROM_FOLLOWED_DOMAIN_INITIAL) {
-            const newFollowedDomains = [...domains]
-            newFollowedDomains[index].user_id_follower = willFollow ? myId : null
+  const route = useRoute();
 
-            FollowingAction.setFollowingDomain(newFollowedDomains, followingDispatch)
-        }
-        if (from === FROM_FOLLOWED_DOMAIN) {
-            const newFollowedDomains = [...followedDomains]
-            newFollowedDomains[index].user_id_follower = willFollow ? myId : null
+  const domains = React.useMemo(() => {
+    return discovery.initialDomains.map((item) => ({
+      ...item,
+      following: item.following !== undefined ? item.following : item.user_id_follower !== null
+    }));
+  }, [discovery.initialDomains]);
+
+  React.useEffect(() => {
+    const parseToken = async () => {
+      const id = await getUserId();
+      if (id) {
+        setMyId(id);
+      }
+    };
+    parseToken();
+  }, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `useEffect` hook is used to fetch the user ID and set it in the state. However, there's no cleanup function provided. If the component unmounts before the async operation finishes, this could lead to a memory leak. Consider adding a cleanup function to cancel the async operation if the component unmounts.

```diff
  React.useEffect(() => {
    let isCancelled = false;
    const parseToken = async () => {
      const id = await getUserId();
      if (!isCancelled && id) {
        setMyId(id);
      }
    };
    parseToken();
    return () => {
      isCancelled = true;
    };
  }, []);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->